backend/uclapi/timetable/management/commands/update_timetable_gencache.py

```diff
@@ -20,41 +21,55 @@ class Command(BaseCommand):
     help = 'Clones timetable related dbs to speed up queries'
 
     def handle(self, *args, **options):
-        classes = [
-            (Module, ModuleA, ModuleB),
-            (Timetable, TimetableA, TimetableB),
-            (Weekstructure, WeekstructureA, WeekstructureB),
-            (Weekmapnumeric, WeekmapnumericA, WeekmapnumericB),
-            (Weekmapstring, WeekmapstringA, WeekmapstringB),
-            (Lecturer, LecturerA, LecturerB),
-            (Rooms, RoomsA, RoomsB),
-            (Sites, SitesA, SitesB),
-            (Students, StudentsA, StudentsB)
+        # Table format: (OracleTable, BucketA, BucketB, HasSetID)
+        tables = [
+            (Module, ModuleA, ModuleB, True),
+            (Timetable, TimetableA, TimetableB, True),
+            (Weekstructure, WeekstructureA, WeekstructureB, True),
+            (Weekmapnumeric, WeekmapnumericA, WeekmapnumericB, True),
+            (Weekmapstring, WeekmapstringA, WeekmapstringB, True),
+            (Lecturer, LecturerA, LecturerB, True),
+            (Rooms, RoomsA, RoomsB, True),
+            (Sites, SitesA, SitesB, True),
+            (Students, StudentsA, StudentsB, True),
+            (Depts, DeptsA, DeptsB, False)
         ]
 
         lock = Lock.objects.all()[0]
-        tbu = 2 if lock.a else 1
+        destination_table_index = 2 if lock.a else 1
+
+        for table_data in tables:
+            print("Inserting contents of {} into {}".format(
+                table_data[0].__name__,
+                table_data[destination_table_index].__name__
+            ))
+            # Only pulls in objects which apply to this year's Set ID
+            if table_data[3]:
```

Comment:
I don't like the way this is handled. Maybe put the filter in the 3rd index? Instead of a boolean?

Reply:
It's not perfect :( My only concern with putting the filter there is that there'd be a lot of duplicated and messy code